core/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -316,6 +316,18 @@ public void writeStringArrayNullable(@Nullable String[] array) throws IOExceptio
         }
     }
 
+    /**
+     * Writes a string array, for nullable string, writes it as 0 (empty string).
+     */
```

Comment:
If it's not used anywhere else (or even if its only used in a couple of places) then I would be +1 to doing it in this PR